### ISSUE TITLE
App crashes if data present in push data

### ISSUE THREAD

**Issue by greywire** [arnesson/cordova-plugin-firebase#76]
No matter what I seem to do, if I pass in 'data' in the push message (as opposed to just 'notification'), the app crashes.

**Comment by robertarnesson** [comment]
platform? callstack/log output please

**Comment by ZameelAmjed** [comment]
Same issue. Platform Android. and it seems two notification callbacks are running in foreground.

**Comment by greywire** [comment]
On further testing I am finding that the app does not crash if I use the Firebase console (or the API) to send a message, but it does crash when I use the Ionic cloud push service to do it. Needless to say I am going to just use the Firebase API directly...

**Comment by ZameelAmjed** [comment]
Firebase console does not send "data" in its json it is "notification" type of push. if i send "notification" through server its passing through but disregarded its data

**Comment by ZameelAmjed** [comment]
App Crash Report
org.apache.cordova.firebase.FirebasePluginMessagingService.sendNotification (FirebasePluginMessagingService.java:75)
org.apache.cordova.firebase.FirebasePluginMessagingService.onMessageReceived (FirebasePluginMessagingService.java:64)
com.google.firebase.messaging.FirebaseMessagingService.zzo ()
com.google.firebase.messaging.FirebaseMessagingService.zzn ()
com.google.firebase.messaging.FirebaseMessagingService.zzm ()
com.google.firebase.iid.zzb$2.run ()
java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1113)
java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:588)
java.lang.Thread.run (Thread.java:818)

**Comment by ZameelAmjed** [comment]
seems message id is not set from API. added following code in FirebasePluginMessagingService.java line 56
```
if(TextUtils.isEmpty(id)){
            Random rand = new Random();
            int  n = rand.nextInt(50) + 1;
            id = Integer.toString(n);
        }
```
Still 2 instances of notification is running

**Comment by zbence** [comment]
That code snippet solves the problem, should put into the project. Thank you

**Comment by robertarnesson** [comment]
699e15b6ee76ac3833e2fda6004196524b5c2a3f

**Comment by zbence** [comment]
Nice but not forget import random

**Comment by robertarnesson** [comment]
fixed in 0.1.18

**Comment by greywire** [comment]
Thank you! App no longer crashes and push notifications working again in my app.
